### PR TITLE
Added unit tests for translator part1

### PR DESCRIPTION
**Pull request by code-incubus**
Signed-off-by: code-incubus <nemanja.milutinovic@comtrade.com>

Added tests for:
- KapuaTranslatorApi
- TranslatorDataMqttKura
- TranslatorResponseMqttKura
- TranslatorDataKuraMqtt 

**Related Issue**
/

**Description of the solution adopted**
This is the first part of tests regarding Translators.

**Screenshots**
/
**Any side note on the changes made**
/

### CODE REVIEW

**Comment by Coduz** [comment]
Converted to draft since it still need some work on that

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  #KapuaTranslatorApi
+
+  Scenario: Finding translator "TranslatorAppCommandKapuaKura"
+  Trying to make translation from CommandRequestMessage to KuraRequestMessage.
+
+    When I try to translate from "org.eclipse.kapua.service.device.management.command.message.internal.CommandRequestMessage" to "org.eclipse.kapua.service.device.call.message.kura.app.request.KuraRequestMessage"
+    Then Translator "TranslatorAppCommandKapuaKura" is found
+    And No exception was thrown
+
+  Scenario: Finding existing translator to null translator
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please change the scenario name to: "Translating CommandRequestMessage to null"

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+###############################################################################
+@unit
+@translator
+
+Feature: Translator Service
+  The Translator Service is responsible for messaging operations between Kura and Kapua.
+
+  #KapuaTranslatorApi
+
+  Scenario: Finding translator "TranslatorAppCommandKapuaKura"
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please change scenario name to: "Translating "CommandRequestMessage" to "KuraRequestMessage""

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+  Scenario: Finding existing translator to null translator
+  Trying to make translation from CommandRequestMessage to null message.
+  NullPointerException should be thrown.
+
+    Given I expect the exception "NullPointerException" with the text "*"
+    When I try to translate from "org.eclipse.kapua.service.device.management.command.message.internal.CommandRequestMessage" to ""
+    Then An exception was thrown
+
+  Scenario: Finding null translator to existing one
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please change scenario name to: "Translating null to KuraRequestMessage"

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+  Trying to make translation from null to KuraRequestMessage message.
+  NullPointerException should be thrown.
+
+    Given I expect the exception "NullPointerException" with the text "*"
+    When I try to translate from "" to "org.eclipse.kapua.service.device.call.message.kura.app.request.KuraRequestMessage"
+    Then An exception was thrown
+
+
+  Scenario: Finding null translator
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please change scenario name to: "Translating empty message to empty message"

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+  Trying to do translation without messages. NPE exception should be thrown.
+
+    Given I expect the exception "NullPointerException" with the text "*"
+    And I try to translate from "" to ""
+    Then An exception was thrown
+
+  Scenario: Finding wrong translator
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please change scenario name to: "Translating from "AssetRequestMessage" to "AssetResponseMessage""

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+  Trying to make translation from AssetRequestMessage to AssetResponseMessage.
+  A translator not found exception should be thrown.
+
+    Given I expect the exception "TranslatorNotFoundException" with the text "*"
+    And I try to translate from "org.eclipse.kapua.service.device.management.asset.message.internal.AssetRequestMessage" to "org.eclipse.kapua.service.device.management.asset.message.internal.AssetResponseMessage"
+    Then An exception was thrown
+
+  #TranslatorDataMqttKura
+
+  Scenario: Translating of mqtt message with invalid payload and valid topic into kura data message
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please  change scenario name to: "Translation of mqtt message with invalid payload and valid topic into kura data message"

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+  Creating mqtt message with invalid payload and valid topic. Trying to translate it into kura data message.
+  Checking if kura data message with valid channel and byte[] payload is received.
+
+    Given I create mqtt message with valid payload "invalidPayload" and valid topic "kapua-sys/rpione3/DEPLOY-V2/GET/packages"
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please change step name to: "Given I create mqtt message with invalid payload "invalidPayload" and valid topic "kapua-sys/rpione3/DEPLOY-V2/GET/packages""

there is a syntax error in the step name, it has to be ...with invalid payload... not "valid".

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+    And I try to translate mqtt message to kura data message
+    Then I got kura data message with "byte[]" payload body
+    And I got kura data message channel with "kapua-sys" and "rpione3" data
+    And No exception was thrown
+
+  Scenario: Translating of mqtt message with valid payload and valid topic into kura data message
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please change scenario name to: "Translation of mqtt message with valid payload and valid topic into kura data message"

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+  Creating mqtt message with valid payload and valid topic. Trying to translate it into kura data message.
+  Checking if mqtt message with valid channel and proper payload metrics is received.
+
+    Given I create mqtt message with valid payload "response.code" and valid topic "kapua-sys/rpione3/DEPLOY-V2/GET/packages"
+    And I try to translate mqtt message to kura data message
+    Then I got kura data message with proper payload metrics
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Can you please make this more specific, checking if proper payload is in the message? 
I mean..it is ok to check if it's not null, but it is also vital to check if the data is as it is supposed to be.

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+    And I got kura data message channel with "kapua-sys" and "rpione3" data
+    And No exception was thrown
+
+  Scenario: Translating of mqtt message with empty payload into kura data message
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please change scenario name to: "Translation of mqtt message with empty payload into kura data message"

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+
+    Given I create mqtt message with empty payload "" and valid topic "kapua-sys/rpione3/DEPLOY-V2/GET/packages"
+    And I try to translate mqtt message to kura data message
+    Then I got kura data message with empty payload
+    And I got kura data message channel with "kapua-sys" and "rpione3" data
+    And No exception was thrown
+
+  #TranslatorResponseMqttKura
+
+  Scenario: Translating of mqtt message with invalid payload and invalid topic into kura response
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please change scenario name to: "Translation of mqtt message with invalid payload and invalid topic into kura response message"

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+  Creating mqtt message with invalid payload and invalid topic. Trying to translate it into kura response.
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please change description to: " Creating mqtt message with invalid payload and invalid topic. Trying to translate it into kura response message."

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 13]
```diff
@@ -0,0 +1,156 @@
...
+
+    Given I expect the exception "InvalidChannelException" with the text "Invalid channel: DEPLOY-V2/GET/packages"
+    And I create mqtt message with invalid payload "invalidPayload" and invalid topic "DEPLOY-V2/GET/packages"
+    When I try to translate mqtt response
+    Then An exception was thrown
+
+  Scenario: Translating of mqtt message with invalid payload and valid topic into kura response
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please change scenario name to: "Translation of mqtt message with invalid payload and valid topic into kura response message"

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 14]
```diff
@@ -0,0 +1,156 @@
...
+  Creating mqtt message with invalid payload and valid topic. Trying to translate it into kura response.
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please change description to: "Creating mqtt message with invalid payload and valid topic. Trying to translate it into kura response message."

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 15]
```diff
@@ -0,0 +1,156 @@
...
+  Check if kura response with byte[] body and correct channel is received.
+
+    Given I create mqtt message with invalid payload "invalidPayload" and valid topic "$EDC/kapua-sys/rpione3/DEPLOY-V2/GET/packages"
+    When I try to translate mqtt response
+    Then I got kura response message with "byte[]" payload body
+    And I got kura response message channel with "GET", "packages", "DEPLOY-V2", "$EDC", "kapua-sys" and "rpione3" data
+    And No exception was thrown
+
+  Scenario: Translating of mqtt message with valid payload and invalid topic into kura response
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please change scenario name to: "Translation of mqtt message with valid payload and invalid topic into kura response message"

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 16]
```diff
@@ -0,0 +1,156 @@
...
+  Creating mqtt message with valid payload and invalid topic. Trying to translate it into kura response.
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please change description to: "Creating mqtt message with valid payload and invalid topic. Trying to translate it into kura response message."

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 17]
```diff
@@ -0,0 +1,156 @@
...
+
+    Given I expect the exception "InvalidChannelException" with the text "Invalid channel: DEPLOY-V2/GET/packages"
+    And I create mqtt message with valid payload "response.code" and invalid topic "DEPLOY-V2/GET/packages"
+    When I try to translate mqtt response
+    Then An exception was thrown
+
+  Scenario: Translating of mqtt message with valid payload and valid topic into kura response
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please change scenario name to: "Translation of mqtt message with valid payload and valid topic into kura response message"

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 18]
```diff
@@ -0,0 +1,156 @@
...
+  Creating mqtt message with valid payload and valid topic. Trying to translate it into kura response.
+  Check if kura response with proper payload, metrics and correct channel is received.
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please change description to: "Creating mqtt message with valid payload and valid topic. Trying to translate it into kura response message.  Check if kura response message with proper payload, metrics and correct channel is received."

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 19]
```diff
@@ -0,0 +1,156 @@
...
+
+    Given I create mqtt message with valid payload "response.code" and valid topic "$EDC/kapua-sys/rpione3/DEPLOY-V2/GET/packages"
+    When I try to translate mqtt response
+    Then I got kura response message with proper payload metrics
+    And I got kura response message channel with "GET", "packages", "DEPLOY-V2", "$EDC", "kapua-sys" and "rpione3" data
+    And No exception was thrown
+
+  #TranslatorDataKuraMqtt
+
+  Scenario: Translating kura data message with valid channel and without body and metrics into mqtt message
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Pleae change scenario to: "Translation of kura data message with valid channel and without body and metrics into mqtt message"

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 20]
```diff
@@ -0,0 +1,156 @@
...
+  Creating kura data message with valid channel and without body and metrics. Trying to translate it into mqtt message.
+  Check if mqtt message with valid topic and empty body is received.
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please change description to: "Checking if mqtt message with valid topic and empty body is received."

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 21]
```diff
@@ -0,0 +1,156 @@
...
+
+    Given I create kura data message with channel with scope "kapua-sys", client id "rpione3" and payload without body and metrics
+    When I try to translate kura data message to mqtt message
+    Then I got mqtt message with channel with scope "kapua-sys", client id "rpione3" and empty body
+    And No exception was thrown
+
+  Scenario: Translating kura data message with valid channel, metrics and without body into mqtt message
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please change scenario name to: "Translation of kura data message with valid channel, metrics and without body into mqtt message"

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 22]
```diff
@@ -0,0 +1,156 @@
...
+  Creating kura data message with valid channel, metrics and without body. Trying to translate it into mqtt message.
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Change description to: "Creating kura data message with valid channel, metrics but without body. Trying to translate it into mqtt message."

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 23]
```diff
@@ -0,0 +1,156 @@
...
+  Check if mqtt message with valid topic and encoded body is received.
+
+    Given I create kura data message with channel with scope "kapua-sys", client id "rpione3" and payload without body and with metrics
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please change the step name to: "Given I create kura data message with channel with scope "kapua-sys", client id "rpione3", valid payload and metrics but without body"

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 24]
```diff
@@ -0,0 +1,156 @@
...
+    When I try to translate kura data message to mqtt message
+    Then I got mqtt message with channel with scope "kapua-sys", client id "rpione3" and non empty body
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Then I get

File: translator/test/src/test/resources/features/TranslatorUnitTests.feature [hunk 25]
```diff
@@ -0,0 +1,156 @@
...
+    And No exception was thrown
+
+  Scenario: Translating kura data message with valid channel, body and metrics into mqtt message
```
Comment (LeoNerdoG) [inline translator/test/src/test/resources/features/TranslatorUnitTests.feature]: Please change the scenario name to: "Translation of kura data message with valid channel, body and metrics into mqtt message"

File: translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java [hunk 26]
```diff
@@ -0,0 +1,363 @@
...
+    }
+
+    @Then("^Translator \"([^\"]*)\" is found$")
+    public void translatorIsFound(String translatorName) {
+        Translator translator = (Translator) stepData.get("Translator");
+        assertEquals(translatorName, translator.getClass().getSimpleName());
+    }
+
+    @Given("^I create mqtt message with (?:valid|invalid|empty) payload \"([^\"]*)\" and (?:valid|invalid) topic \"([^\"]*)\"$")
+    public void iTryToProvideInvalidTopic(String payload, String topic) {
```
Comment (LeoNerdoG) [inline translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java]: Please change method name to: "public void creatingMqttMessage(String payload, String topic)" becasue you have 6 options here and not only one.

File: translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java [hunk 27]
```diff
@@ -0,0 +1,363 @@
...
+
+    @Then("^I got kura response message with \"([^\"]*)\" payload body$")
+    public void iGotKuraResponseMessageWithPayloadBody(String payloadType) {
+        KuraResponseMessage kuraResponseMessage = (KuraResponseMessage) stepData.get("KuraResponseMessage");
+
+        assertTrue(kuraResponseMessage.getPayload().getBody().getClass().getSimpleName().equals(payloadType));
+    }
+
+    @Then("^I got kura response message with proper payload metrics$")
+    public void iGotKuraResponseMessageWithPayloadAndChannelWithAndData() {
```
Comment (LeoNerdoG) [inline translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java]: Plesase change the method name to: "public voidkuraResponseMessageWithPayloadAndChannelAndData()"

File: translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java [hunk 28]
```diff
@@ -0,0 +1,363 @@
...
+
+    @And("^I try to translate kura data message to mqtt message$")
+    public void iTryToTranslateKuraDataMessageToMqttMessage() throws Exception {
+        KuraDataMessage kuraDataMessage = (KuraDataMessage) stepData.get("KuraDataMessage");
+        MqttMessage mqttMessage = translatorDataKuraMqtt.translate(kuraDataMessage);
+        stepData.put("MqttMessage", mqttMessage);
+    }
+
+    @Then("^I got mqtt message with channel with scope \"([^\"]*)\", client id \"([^\"]*)\" and (?:empty body|non empty body)$")
+    public void iGotMqttMessageWithChannelWithClientIdScopeAndWithoutMetricsAndBodyInPayload(String scope, String clientId) {
```
Comment (LeoNerdoG) [inline translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java]: Please change method name to: "mqttMessageWithChanneScopeClienIDandBody()" and even this is too long... you can add a comment bellow, but the method name should never be this long.

File: translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java [hunk 29]
```diff
@@ -0,0 +1,363 @@
...
+
+        if (mqttMessage.getPayload().getBody().length == 0) {
+            assertTrue(mqttMessage.getPayload().getBody().length == 0);
+        } else {
+            assertTrue(mqttMessage.getPayload().getBody().length != 0);
+        }
+    }
+
+    @And("^I got kura response message channel with \"(.+)\", \"(.+)\", \"(.+)\", \"(.+)\", \"(.+)\" and \"(.+)\" data$")
+    public void iGotKuraResponseMessageChannelWithAndData(String replyPart, String requestId, String appId, String messageClassification, String scope, String clientId) {
```
Comment (LeoNerdoG) [inline translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java]: Change method name to: "kuraResponseMessageWithChannelAndData"

File: translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java [hunk 30]
```diff
@@ -0,0 +1,363 @@
...
+        try {
+            KuraResponseMessage kuraResponseMessage = translatorResponseMqttKura.translate(mqttMessage);
+            stepData.put("KuraResponseMessage", kuraResponseMessage);
+        } catch (KapuaException ex) {
+            verifyException(ex);
+        }
+    }
+
+    @Then("^I got kura response message with \"([^\"]*)\" payload body$")
+    public void iGotKuraResponseMessageWithPayloadBody(String payloadType) {
```
Comment (LeoNerdoG) [inline translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java]: Change method name to: "kuraResponseMessageWithPayloadBody"

File: translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java [hunk 31]
```diff
@@ -0,0 +1,363 @@
...
+        assertTrue(kuraResponseMessage.getChannel().getReplyPart().equals(replyPart));
+        assertTrue(kuraResponseMessage.getChannel().getRequestId().equals(requestId));
+        assertTrue(kuraResponseMessage.getChannel().getAppId().equals(appId));
+        assertTrue(kuraResponseMessage.getChannel().getMessageClassification().equals(messageClassification));
+        assertTrue(kuraResponseMessage.getChannel().getScope().equals(scope));
+        assertTrue(kuraResponseMessage.getChannel().getClientId().equals(clientId));
+    }
+
+    @Given("^I create kura data message with channel with scope \"([^\"]*)\", client id \"([^\"]*)\" and payload without body and with metrics$")
+    public void iCreateKuraDataMessageWithChannelWithScopeClientIdAndPayloadWithoutBodyAndWithMetrics(String scope, String clientId) {
```
Comment (LeoNerdoG) [inline translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java]: Change method name to: "kuraDataMessageWithoutBodyAndMetrics"

File: translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java [hunk 32]
```diff
@@ -0,0 +1,363 @@
...
+        KuraDataChannel kuraDataChannel = new KuraDataChannel(scope, clientId);
+        KuraDataPayload kuraDataPayload = new KuraDataPayload();
+        kuraDataPayload.getMetrics().put("response.code", 200);
+        KuraDataMessage kuraDataMessage = new KuraDataMessage(kuraDataChannel, date, kuraDataPayload);
+
+        stepData.put("KuraDataMessage", kuraDataMessage);
+    }
+
+    @Given("^I create kura data message with channel with scope \"([^\"]*)\", client id \"([^\"]*)\" and payload with body and metrics$")
+    public void iCreateKuraDataMessageWithChannelWithScopeClientIdAndPayloadWithBodyAndMetrics(String scope, String clientId) {
```
Comment (LeoNerdoG) [inline translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java]: Change method name to: "fullKuraDataMessage"

File: translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java [hunk 33]
```diff
@@ -0,0 +1,363 @@
...
+
+    @And("^I try to translate kura data message to jms message$")
+    public void iTryToTranslateKuraDataMessageToJmsMessage() throws Exception {
+        KuraDataMessage kuraDataMessage = (KuraDataMessage) stepData.get("KuraDataMessage");
+        JmsMessage jmsMessage = translatorDataKuraJms.translate(kuraDataMessage);
+        stepData.put("JmsMessage", jmsMessage);
+    }
+
+    @Then("^I got kura data message channel with \"([^\"]*)\" scope, \"([^\"]*)\" client id and proper semanticPart$")
+    public void iGotKuraDataMessageChannelWithScopeClientIdAndProperSemanticPart(String scope, String clientId, List<String> semanticParts) {
```
Comment (LeoNerdoG) [inline translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java]: Change method name to: "iCreateJmsMessageWithInvalidPayloadAndTopic"

File: translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java [hunk 34]
```diff
@@ -0,0 +1,363 @@
...
+        assertEquals(scope, kuraDataMessage.getChannel().getScope());
+        assertEquals(clientId, kuraDataMessage.getChannel().getClientId());
+
+        for (String semanticPart : semanticParts) {
+            assertTrue(kuraDataMessage.getChannel().getSemanticParts().contains(semanticPart));
+        }
+    }
+
+    @Then("^I got jms message with topic \"([^\"]*)\" and (?:empty body|non empty body)$")
+    public void iGotJmsMessageWithTopicAndEmptyPayloa(JmsTopic topic) {
```
Comment (LeoNerdoG) [inline translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java]: Change method name to: "iGotJmsMessageWithTopicAndEmptyPayload"

File: translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java [hunk 35]
```diff
@@ -0,0 +1,363 @@
...
+        KuraResponseMessage kuraResponseMessage = (KuraResponseMessage) stepData.get("KuraResponseMessage");
+
+        assertTrue(kuraResponseMessage.getPayload().getBody().getClass().getSimpleName().equals(payloadType));
+    }
+
+    @Then("^I got kura response message with proper payload metrics$")
+    public void iGotKuraResponseMessageWithPayloadAndChannelWithAndData() {
+        KuraResponseMessage kuraResponseMessage = (KuraResponseMessage) stepData.get("KuraResponseMessage");
+
+        assertTrue(kuraResponseMessage.getPayload().getMetrics() != null);
```
Comment (LeoNerdoG) [inline translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java:198]: Please change this to assertEquals(), becasue compartin to "!=null" is not enough.

File: translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java [hunk 36]
```diff
@@ -0,0 +1,363 @@
...
+        Date date = new Date();
+        MqttTopic mqttTopic = new MqttTopic(topic);
+
+        KuraPayload kuraPayload = new KuraPayload();
+        if (payload.equals("invalidPayload") || payload.equals("")) {
+            kuraPayload.setBody(payload.getBytes());
+        } else {
+            kuraPayload.getMetrics().put(payload, 200);
+        }
+
```
Comment (LeoNerdoG) [inline translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java:168]: insert try/catch block, the "stepdata.put" part can be left ouside the try/catch.

File: translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java [hunk 37]
```diff
@@ -0,0 +1,363 @@
...
+        assertTrue(kuraResponseMessage.getPayload().getMetrics() != null);
+    }
+
+    @Given("^I create kura data message with channel with scope \"([^\"]*)\", client id \"([^\"]*)\" and payload without body and metrics$")
+    public void iCreateKuraDataMessage(String scope, String clientId) {
+        KuraDataChannel kuraDataChannel = new KuraDataChannel(scope, clientId);
+        Date date = new Date();
+        KuraDataPayload kuraDataPayload = new KuraDataPayload();
+        KuraDataMessage kuraDataMessage = new KuraDataMessage(kuraDataChannel, date, kuraDataPayload);
```
Comment (LeoNerdoG) [inline translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java]: insert try/catch block, the "stepdata.put" part can be left ouside the try/catch.

File: translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java [hunk 38]
```diff
@@ -0,0 +1,363 @@
...
+        stepData.put("KuraDataMessage", kuraDataMessage);
+    }
+
+    @And("^I try to translate kura data message to mqtt message$")
+    public void iTryToTranslateKuraDataMessageToMqttMessage() throws Exception {
+        KuraDataMessage kuraDataMessage = (KuraDataMessage) stepData.get("KuraDataMessage");
+        MqttMessage mqttMessage = translatorDataKuraMqtt.translate(kuraDataMessage);
```
Comment (LeoNerdoG) [inline translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java]: insert try/catch block, the "stepdata.put" part can be left ouside the try/catch.

File: translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java [hunk 39]
```diff
@@ -0,0 +1,363 @@
...
+    public void iGotMqttMessageWithChannelWithClientIdScopeAndWithoutMetricsAndBodyInPayload(String scope, String clientId) {
+        MqttMessage mqttMessage = (MqttMessage) stepData.get("MqttMessage");
+
+        String requestTopic = scope.concat("/" + clientId);
+        assertEquals(requestTopic, mqttMessage.getRequestTopic().getTopic());
+
+        if (mqttMessage.getPayload().getBody().length == 0) {
+            assertTrue(mqttMessage.getPayload().getBody().length == 0);
+        } else {
+            assertTrue(mqttMessage.getPayload().getBody().length != 0);
```
Comment (LeoNerdoG) [inline translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java:227]: can you check more specific and not only that is !=0?

File: translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java [hunk 40]
```diff
@@ -0,0 +1,363 @@
...
+        assertTrue(kuraResponseMessage.getChannel().getClientId().equals(clientId));
+    }
+
+    @Given("^I create kura data message with channel with scope \"([^\"]*)\", client id \"([^\"]*)\" and payload without body and with metrics$")
+    public void iCreateKuraDataMessageWithChannelWithScopeClientIdAndPayloadWithoutBodyAndWithMetrics(String scope, String clientId) {
+        Date date = new Date();
+        KuraDataChannel kuraDataChannel = new KuraDataChannel(scope, clientId);
+        KuraDataPayload kuraDataPayload = new KuraDataPayload();
+        kuraDataPayload.getMetrics().put("response.code", 200);
+        KuraDataMessage kuraDataMessage = new KuraDataMessage(kuraDataChannel, date, kuraDataPayload);
```
Comment (LeoNerdoG) [inline translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java]: insert try/catch block, the "stepdata.put" part can be left ouside the try/catch.

File: translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java [hunk 41]
```diff
@@ -0,0 +1,363 @@
...
+    }
+
+    @Given("^I create kura data message with channel with scope \"([^\"]*)\", client id \"([^\"]*)\" and payload with body and metrics$")
+    public void iCreateKuraDataMessageWithChannelWithScopeClientIdAndPayloadWithBodyAndMetrics(String scope, String clientId) {
+        Date date = new Date();
+        KuraDataChannel kuraDataChannel = new KuraDataChannel(scope, clientId);
+        KuraDataPayload kuraDataPayload = new KuraDataPayload();
+        kuraDataPayload.setBody("Payload".getBytes());
+        kuraDataPayload.getMetrics().put("response.code", 200);
+        KuraDataMessage kuraDataMessage = new KuraDataMessage(kuraDataChannel, date, kuraDataPayload);
```
Comment (LeoNerdoG) [inline translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java]: insert try/catch block, the "stepdata.put" part can be left ouside the try/catch.

File: translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java [hunk 42]
```diff
@@ -0,0 +1,363 @@
...
+
+        stepData.put("KuraDataMessage", kuraDataMessage);
+    }
+
+    @Given("^I try to translate mqtt message to kura data message$")
+    public void iTryToTranslateMqttMessageToKuraMessage() throws Exception {
+        MqttMessage mqttMessage = (MqttMessage) stepData.get("MqttMessage");
+        KuraDataMessage kuraDataMessage = translatorDataMqttKura.translate(mqttMessage);
```
Comment (LeoNerdoG) [inline translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java]: insert try/catch block, the "stepdata.put" part can be left ouside the try/catch.

File: translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java [hunk 43]
```diff
@@ -0,0 +1,363 @@
...
+        Date date = new Date();
+        JmsTopic jmsTopic = new JmsTopic(topic);
+
+        KuraPayload kuraPayload = new KuraPayload();
+        if (payload.equals("invalidPayload") || payload.equals("")) {
+            kuraPayload.setBody(payload.getBytes());
+        } else {
+            kuraPayload.getMetrics().put(payload, 200);
+        }
+
```
Comment (LeoNerdoG) [inline translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java:313]: insert try/catch block, the "stepdata.put" part can be left ouside the try/catch.

File: translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java [hunk 44]
```diff
@@ -0,0 +1,363 @@
...
+            stepData.put("KuraDataMessage", kuraDataMessage);
+        } catch (KapuaException ex) {
+            verifyException(ex);
+        }
+    }
+
+    @And("^I try to translate kura data message to jms message$")
+    public void iTryToTranslateKuraDataMessageToJmsMessage() throws Exception {
+        KuraDataMessage kuraDataMessage = (KuraDataMessage) stepData.get("KuraDataMessage");
+        JmsMessage jmsMessage = translatorDataKuraJms.translate(kuraDataMessage);
```
Comment (LeoNerdoG) [inline translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java]: insert try/catch block, the "stepdata.put" part can be left ouside the try/catch.

**Review by LeoNerdoG — CHANGES_REQUESTED** [review]
Hey @code-incubus,
I have finally reviewed this branch and there are some changes to be made. 
Plase address them and then push the changes.

**Review by LeoNerdoG — APPROVED** [review]
@code-incubus I have checked the changes and they look OK. 
@coduz - can you please merge this PR since Travis and all tests have successfully passed?

**Review by Coduz — APPROVED** [review]

**Pull request merged**